test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -116,58 +140,30 @@ TEST_F(XdsOverrideHostTest, OverrideHost) {
 }
 
 TEST_F(XdsOverrideHostTest, OverrideHostSubchannelNotFound) {
-  const std::array<absl::string_view, 2> kAddresses = {"ipv4:127.0.0.1:441",
-                                                       "ipv4:127.0.0.1:442"};
-  EXPECT_EQ(ApplyUpdate(BuildUpdate(kAddresses,
-                                    MakeXdsOverrideHostConfig("round_robin")),
-                        policy_.get()),
-            absl::OkStatus());
-  ExpectConnectingUpdate();
-  // Ready up both subchannels
-  for (auto address : kAddresses) {
-    auto subchannel = FindSubchannel({address});
-    ASSERT_NE(subchannel, nullptr);
-    ASSERT_TRUE(subchannel->ConnectionRequested());
-    subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
-    subchannel->SetConnectivityState(GRPC_CHANNEL_READY);
-  }
-  WaitForConnected();
-  ExpectState(GRPC_CHANNEL_READY);
-  auto picker = ExpectState(GRPC_CHANNEL_READY);
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto maybe_picker = InitRoundRobin(kAddresses);
+  ASSERT_TRUE(maybe_picker.has_value());
+  auto picker = *maybe_picker;
   ASSERT_NE(picker, nullptr);
-  // Make sure child policy works
-  EXPECT_NE(ExpectPickComplete(picker.get()), ExpectPickComplete(picker.get()));
-  // Check that the host is overridden
   std::map<UniqueTypeName, std::string> call_attributes{
       {XdsHostOverrideTypeName(), "no such host"}};
-  std::unordered_set<absl::optional<std::string>> picks{
-      ExpectPickComplete(picker.get(), call_attributes),
-      ExpectPickComplete(picker.get(), call_attributes)};
+  std::unordered_set<std::string> picks;
```

Comment:
Instead of doing this by hand, I suggest adding a `call_attributes` parameter to `ExpectRoundRobinPicks()` and then just use that method here.